compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -123,7 +133,12 @@ bool isRISCV(IREE::HAL::ExecutableTargetAttr targetAttr) {
 }
 
 bool isVMVXBackend(IREE::HAL::ExecutableTargetAttr targetAttr) {
-  return targetAttr.getBackend().getValue().startswith("vmvx");
+  return targetAttr && targetAttr.getBackend().getValue().startswith("vmvx");
+}
+
+bool hasMicrokernels(IREE::HAL::ExecutableTargetAttr targetAttr) {
+  auto enableMicrokernels = getConfigBoolAttr(targetAttr, "ukernels");
+  return enableMicrokernels && enableMicrokernels->getValue();
```

Comment:
nit: use `.value()`

I recently notice that there is a plan about deprecating `Optional<X>::hasValue/getValue/getValueOr`. I think we should start using the new style.

https://discourse.llvm.org/t/deprecating-llvm-optional-x-hasvalue-getvalue-getvalueor/63716/15

Reply:
Ah, but here in the expression `enableMicrokernels->getValue()`, it is the `operator->` that queries the `Optional::value()` already, returning a `BoolAttr`, so the `getValue()` call here is `BoolAttr::getValue`. I checked if `BoolAttr` was undergoing a similar renaming but it seems not to:

```
/usr/local/google/home/benoitjacob/iree/compiler/src/iree/compiler/Codegen/Utils/Utils.cpp:141:52: error: no member named 'value' in 'mlir::BoolAttr'
  return enableMicrokernels && enableMicrokernels->value();
```

Reply:
@MaheshRavishankar this is an example of a conversation that has been had due to the nesting of attributes in `Optional` :wink:   (insider joke / see #11301)